### PR TITLE
adding sagemaker support | notebook-instance

### PR DESCRIPTION
**Pull request by joshuaroot**
newly released service 'SageMaker' has no support in custodian. 
adding support for the 'notebook-instance'
- start, stop, delete actions
  - adding state transition filter (from ec2)
- tag actions & filters

### CODE REVIEW

**Comment by kapilt** [comment]
do notebook instances show up directly as ec2 instances, if so do they have a tag that discriminates they are sagemaker notebook instances?

File: c7n/resources/sagemaker.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+from c7n.utils import local_session, type_schema
+from c7n.actions import BaseAction
+from c7n.tags import RemoveTag, Tag, TagActionFilter, TagDelayedAction
+
+
+filters = FilterRegistry('sagemaker.filters')
+filters.register('marked-for-op', TagActionFilter)
+
+
+@resources.register('notebook-instance')
```
Comment (kapilt) [inline c7n/resources/sagemaker.py]: notebook feels a little ambigious in a flat namespace of resource types, how about sagemaker-instance instead.

Comment (kapilt) [reply c7n/resources/sagemaker.py]: actually how about `sagemaker-notebook`

Comment (joshuaroot) [reply c7n/resources/sagemaker.py]: that sounds good... also helps clarify the resource... notebook-instance is a little vague

**Pull request merged**